app/models/existing_role.rb

```
@@ -0,0 +1,34 @@
+class ExistingRole < ActiveRecord::Base
+  validates :name, presence: true
+  def self.update_resource(param)
+    id = param[:id]
+    name = param[:name]
+    role = where("id = ?", id).first
+    if role
+      role.update_attribute("name", name)
+      response = {
+        status: 200,
+        message: "Updated successfully"
+      }
+    else
+      response = {
```

Just return the error or throw instead of returning a response with http code.